src/cicd/pipelines.ts

```diff
@@ -156,6 +173,40 @@ export class CICDPipelineStack extends Stack {
     return this;
   }
 
+  addWave(props: AddApplicationWaveProps) {
+    /*
+    Add multiple application stages in parallel to the CICD pipeline. This stage deploys your application infrastructure.
+      Parameters
+    ----------
+    stageId: str
+    Identifier of the wave
+    stages: Stage[]
+    Application stage instance
+    manualApprovals: Optional[bool]
+    Configure manual approvals. False by default
+      Returns
+    -------
+    pipeline : CICDPipelineStack
+    CICDPipelineStack
+    */
+    if (this.pipeline === undefined) {
+      throw new Error('`.buildPipeline()` needs to be called first before adding application stages to the pipeline.');
+    }
+    var manualApprovals = props.manualApprovals ?? false; // || this._config.get_env_config(stage_id).get('manual_approvals');
+
+    var wave = new Wave(props.stageId);
+    if (manualApprovals) {
+      wave.addPre(new ManualApprovalStep('PromoteTo' + toTitleCase(props.stageId)));
+    }
+
+    Object.entries(props.stages).forEach(([_, value]) => {
```

Comment:
I'm not sure I understand the use for `Object.entries` here. Shouldn't we just use:
```javascript
props.stages.forEach((stage) => {
  ...
});
```

Reply:
I tried this, but was getting `Property 'forEach' does not exist on type 'Stage'.ts(2339)`

Reply:
Oops, had `Stage` not `Stage[]` in my props